griptape/tools/openweather_client/tool.py

```diff
@@ -0,0 +1,147 @@
+        if coordinates:
+            lat, lon = coordinates
+            request_params = {
+                'lat': lat,
+                'lon': lon,
+                'exclude': 'minutely,hourly,daily,alerts',
+                'appid': self.api_key,
+                'units': self.units
+            }
+            return self._fetch_weather_data(request_params)
+        else:
+            return ErrorArtifact(f"Error fetching coordinates for location: {location}")
+
+    @activity(
+        config={
+            "description": "Fetch hourly forecast for a given location up to 48 hours ahead using the OpenWeather API. Temperatures are returned in {{ units }} by default.",
+            "schema": Schema({
+                Literal("location", description="Location to fetch hourly forecast for."): str
+            }),
+        }
+    )
+    def get_hourly_forecast_by_location(self, params: dict) -> ListArtifact | TextArtifact | ErrorArtifact:
+        location = params["values"].get("location")
+        coordinates = self._fetch_coordinates(location)
+        if coordinates:
+            lat, lon = coordinates
+            request_params = {
+                'lat': lat,
+                'lon': lon,
+                'exclude': 'minutely,current,daily,alerts',
+                'appid': self.api_key,
+                'units': self.units
+            }
+            return self._fetch_weather_data(request_params)
+        else:
+            return ErrorArtifact(f"Error fetching coordinates for location: {location}")
+
+    @activity(
+        config={
+            "description": "Fetch daily forecast for a given location up to 8 days ahead using the OpenWeather API. Temperatures are returned in {{ units }} by default.",
+            "schema": Schema({
+                Literal("location", description="Location to fetch daily forecast for."): str
+            }),
+        }
+    )
+    def get_daily_forecast_by_location(self, params: dict) -> ListArtifact | TextArtifact | ErrorArtifact:
+        location = params["values"].get("location")
+        coordinates = self._fetch_coordinates(location)
+        if coordinates:
+            lat, lon = coordinates
+            request_params = {
+                'lat': lat,
+                'lon': lon,
+                'exclude': 'minutely,hourly,current,alerts',
+                'appid': self.api_key,
+                'units': self.units
+            }
+            return self._fetch_weather_data(request_params)
+        else:
+            return ErrorArtifact(f"Error fetching coordinates for location: {location}")
+
+    def _fetch_weather_data(self, request_params: dict) -> ListArtifact | TextArtifact | ErrorArtifact:
+        try:
+            response = requests.get(self.BASE_URL, params=request_params)
+            if response.status_code == 200:
+                data = response.json()
+                if isinstance(data, list):
+                    return ListArtifact(data)
```

Comment:
Do a list comprehension to wrap results in `InfoArtifact` for `data` as `ListArtifact` expects a list of artifacts to be passed.